src/compression/snappy.rs

```diff
@@ -64,11 +64,26 @@ impl<B: ByteBuf> Decompressor<B> for Snappy {
         F: FnOnce(&mut Self::Buf) -> Result<R>,
     {
         // See https://github.com/xerial/snappy-java?tab=readme-ov-file#compatibility-notes
-        let magic = compressed
+        if !compressed.has_remaining() {
+            anyhow::bail!("expected some bytes in snappy stream");
+        }
+
+        // We fall back to non-Kafka "raw" snappy compression if the magic header is not present
+        // just like the [Java implementation](https://github.com/xerial/snappy-java/blob/48b31663c8b9d01d758368a26416ef6194045c5f/src/main/java/org/xerial/snappy/SnappyInputStream.java#L114).
+        if compressed
             .try_get_bytes(MAGIC_HEADER.len())
-            .context("not enough bytes for magic header")?;
-        if *magic != MAGIC_HEADER[..] {
-            anyhow::bail!("invalid snappy magic header");
+            .ok()
+            .is_none_or(|magic| *magic != MAGIC_HEADER[..])
+        {
+            let compressed = compressed.copy_to_bytes(compressed.remaining());
+            let actual_len = decompress_len(&compressed).context("Failed to decompress snappy")?;
+            let mut tmp = BytesMut::new();
+            tmp.resize(actual_len, 0);
```

Comment:
This will pad actual_len with `0`s if the buffer is too small. I believe this could happen if we havent received the full message yet (unless theres some other frame wrapping this one outside of this logic)

We should be able to avoid this by using try_get_bytes like we do for the kafka specific snappy implementation.

Reply:
I'm not sure I get the situation, `tmp` is a new buffer and we know the exact decompressed length, right? So all the zeroes will be overwritten. Where do you suggest using `try_get_bytes` exactly?

Reply:
My bad, I thought we were resizing the input buffer but we are actually resizing the empty output buffer.

Can we swap it to use https://docs.rs/bytes/latest/bytes/struct.BytesMut.html#method.zeroed
That will make this clearer.